topic4/contracts/Math.sol

```diff
@@ -3,5 +3,48 @@ pragma solidity ^0.4.23;
 import './interfaces/IMath.sol';
 
 contract Math is IMath {
-		
+
+    uint _stateNumber = 0;
+
+	function stateNumber() external view returns (uint) {
+        return _stateNumber;
+    }
+
+	function resetStateNumber() external {
+        _stateNumber = 0;
+    }
+
+	function add(uint256 _inputNumber) external {
+        uint sum = _stateNumber + _inputNumber;
+		assert(sum >= _stateNumber);
+		_stateNumber = sum;
+    }
+
+	function sub(uint256 _inputNumber) external {
+		assert(_inputNumber <= _stateNumber);
+        uint diff = _stateNumber - _inputNumber;
+		_stateNumber = diff;
+    }
+
+	function mul(uint256 _inputNumber) external {
+        uint product = _stateNumber * _inputNumber;
+		assert(_stateNumber == 0 || _stateNumber == product / _inputNumber);
```

Comment:
6. Use the linter - navigate to the project folder and run `solium -d contracts/ --fix`.
7. What happens if the `_stateNumber` is 0 and the `_inputNumber` is 0?